tasks/src/main/java/ru/sberbank/school/task02/calculator/ExtendedCurrencyCalculator.java

```diff
@@ -0,0 +1,110 @@
+        BigDecimal amountQuote;
+        List<Quote> finalQuoteList = new ArrayList<>();
+        for (Quote quote: quotes) {
+            if (quote.getVolume().isInfinity() || quote.getVolumeSize().compareTo(amountOfRequest) > 0) {
+                finalQuoteList.add(quote);
+            }
+        }
+
+        List<Quote> sortedQuoteList = filterQutesList(comparator, finalQuoteList);
+        for (Quote quote : sortedQuoteList) {
+            amountQuote = quote.getVolumeSize();
+            if (amountQuote.compareTo(amountOfRequest) > 0) {
+                return Optional.of(quote);
+            }
+            if (quote.getVolume().isInfinity()) {
+                return Optional.of(quote);
+            }
+        }
+        return Optional.empty();
+    }
+
+    List<Quote> filterQutesList(Comparator<Quote> comparator, List<Quote> quotes) {
+        List<Quote> filterBySymbolList = quotes.stream()
+                .sorted(comparator)
+                .collect(Collectors.toList());
+        System.out.println("Sorted List: ");
+        showQuotes(filterBySymbolList);
+        return filterBySymbolList;
+    }
+
+
+    @Override
+    public Optional<BigDecimal> convertReversed(ClientOperation operation, Symbol symbol, BigDecimal amount, Beneficiary beneficiary) {
+        quotes = externalQuotesService.getQuotes(Symbol.USD_RUB);
+        this.symbolOfRequest = symbol;
+        this.amountOfRequest = amount;
+        System.out.println("Get benificiary: " + beneficiary.toString());
+        System.out.println("Get symbol: " + symbol.getSymbol());
+        System.out.println("Get client operation: " + operation.toString());
+        System.out.println("Get request volume: " +  amountOfRequest );
+        showQuotes(quotes);
+//        if ((beneficiary == Beneficiary.BANK && operation == ClientOperation.BUY)
+//                || (beneficiary == Beneficiary.CLIENT && operation == ClientOperation.SELL)) {
+//            rounding_mode = RoundingMode.HALF_UP;
+//        }
+//        if ((beneficiary == Beneficiary.BANK && operation == ClientOperation.SELL)
+//                || (beneficiary == Beneficiary.CLIENT && operation == ClientOperation.BUY)) {
+//            rounding_mode = BigDecimal.HALF_UP;
+//        }
+        this.amountOfRequest = amount.setScale(10,  rounding_mode);
+
+        if (check()) {
+            return extendedOperation(operation, beneficiary);
+        }
+        System.out.println("Return Optional.empty() after check");
+        return Optional.empty();
+    }
+
+    @Override
+    public Optional<BigDecimal> convertReversed(ClientOperation operation,
+                                                Symbol symbol,
+                                                BigDecimal amount,
+                                                double delta,
+                                                Beneficiary beneficiary) {
+         return Optional.empty();
```

Comment:
&#39;method def&#39; child has incorrect indentation level 9, expected level should be 8.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1576066157>*